digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/DigitalTwinDeviceClient.java

```diff
@@ -109,166 +99,284 @@
                     }
-                })
-                .singleOrError();
+                }).singleOrError();
     }
 
     /**
-     * Registers the specified {@link AbstractDigitalTwinInterfaceClient} with the DigitalTwin Service.
+     * Trigger properties sync up with the DigitalTwin Service.
      * The call will be blocked and will return the result once registration is processed.
-     * It must not be called multiple times. If a given Digital Twin device needs to have its handles re-registered, it needs to create a new DigitalTwinDeviceClient instance.
      *
-     * @param deviceCapabilityModelId     Device Capability Model Id
-     * @param digitalTwinInterfaceClients An list of {@link AbstractDigitalTwinInterfaceClient}s to register with the service.
      * @return Result of this sync function.
      */
-    public DigitalTwinClientResult registerInterfaces(@NonNull final String deviceCapabilityModelId,
-            @NonNull final List<? extends AbstractDigitalTwinInterfaceClient> digitalTwinInterfaceClients) {
-        return registerInterfacesAsync(deviceCapabilityModelId, digitalTwinInterfaceClients).blockingGet();
+    public DigitalTwinClientResult syncupProperties() {
+        return syncupPropertiesAsync().blockingGet();
+    }
+
+    /**
+     * Application should call this function to notify all components they are now ready to use. All bound component function {@link AbstractDigitalTwinComponent#ready()} will be triggered.
+     *
+     * @return Failure if no component was bound, success otherwise.
+     */
+    public DigitalTwinClientResult ready() {
+        synchronized (lock) {
+            if(components.isEmpty()) {
+                return DIGITALTWIN_CLIENT_ERROR_COMPONENTS_NOT_BOUND;
+            }
+
+            for (AbstractDigitalTwinComponent component : components.values()) {
+                component.ready();
+            }
+        }
+        return DIGITALTWIN_CLIENT_OK;
+    }
+
+    Flowable<DigitalTwinClientResult> sendTelemetryAsync(
+            @NonNull final String digitalTwinComponentName,
+            @NonNull final String telemetryName,
+            @NonNull final String payload) {
+        return ensureConnectAsync()
+                .flatMap(new Function<DigitalTwinClientResult, Flowable<DigitalTwinClientResult>>() {
+                    @Override
+                    public Flowable<DigitalTwinClientResult> apply(DigitalTwinClientResult result) {
+                        return Flowable.create(new FlowableOnSubscribe<DigitalTwinClientResult>() {
+                            @Override
+                            public void subscribe(FlowableEmitter<DigitalTwinClientResult> emitter) throws Throwable {
+                                log.debug("Sending TelemetryAsync...");
+                                SimpleEntry body = new SimpleEntry<>(telemetryName, new JsonRawValue(payload));
+                                Message message = new Message(serialize(body));
```

Comment:
what about content encoding and type?

Reply:
Default charset for JAVA is `UTF-8`. If we'd like to add default charset or content type, it should be handled in device SDK rather than digital twin SDK.

Reply:
I don't know about defaults. In C# and C, the sender is supposed to specify both properties. If one doesn't, then the messages cannot be processed by the hub's rules engine. That means routing is broken.

This has nothing to do with a language's default character set, and telling the hub how to interpret the message, as sent.

Reply:
done.